scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -280,6 +280,7 @@ pub struct DetailedTomlDependency {
 pub struct TomlTarget<P> {
     pub name: Option<SmolStr>,
     pub source_path: Option<Utf8PathBuf>,
+    pub group_id: Option<SmolStr>,
```

Comment:
Hmm I cannot imagine how we should deal with users manually specifying `group_id` in their tomls. Do we have to expose this property here and not just make it a private thing?

Reply:
Hmm, I don't see why this would be a problem? If users asks us to compile a set of targets simultaneously and this is possible, then why not? 🤔

Reply:
1. I'm just afraid of making this "optimization" an actual API
2. I do not recall a check that valid `lib.cairo` is actually constructible for the group. Or does it exist somewhere?

Reply:
2. It should be validated in `CompilationUnitComponent::try_new`

Reply:
1. Ok,  I will disallow this then